ndindex/tests/test_ndindex.py

```diff
@@ -1,17 +1,34 @@
+import ast
 import inspect
-
 import numpy as np
-
-from hypothesis import given, example, settings
-
+from hypothesis import example, given, settings
+from hypothesis.strategies import one_of
 from pytest import raises, warns
 
-from ..ndindex import ndindex, asshape
+from ..ndindex import ndindex, parse_index, asshape
 from ..integer import Integer
 from ..ellipsis import ellipsis
 from ..integerarray import IntegerArray
 from ..tuple import Tuple
-from .helpers import ndindices, check_same, assert_equal
+from .helpers import ndindices, check_same, assert_equal, ellipses, ints, slices, tuples, _doesnt_raise
+
+Tuples = tuples(one_of(
+    ellipses(),
+    ints(),
+    slices(),
+)).filter(_doesnt_raise)
+
+ndindexStrs = one_of(
+    ellipses(),
+    ints(),
+    slices(),
+    Tuples,
+).map(lambda x: f'{x}')
```

Comment:
Put strategies in helpers.py

Reply:
You can include arrays here too, although you'll have to use the repr form in that case.